src/Compilers/Core/Portable/PEWriter/PeWriter.cs

```diff
@@ -85,13 +85,26 @@ internal sealed class PeWriter
             bool deterministic,
             CancellationToken cancellationToken)
         {
-            // If PDB writer is given, we have to have PDB path.
-            Debug.Assert(nativePdbWriterOpt == null || pdbPathOpt != null);
+                // If PDB writer is given, we have to have PDB path.
+                Debug.Assert(nativePdbWriterOpt == null || pdbPathOpt != null);
 
-            var peWriter = new PeWriter(context.Module.Properties, context.Module.Win32Resources, context.Module.Win32ResourceSection, pdbPathOpt, deterministic);
-            var mdWriter = FullMetadataWriter.Create(context, messageProvider, allowMissingMethodBodies, deterministic, getPortablePdbStreamOpt != null, cancellationToken);
-
-            return peWriter.WritePeToStream(mdWriter, getPeStream, getPortablePdbStreamOpt, nativePdbWriterOpt);
+            try
```

Comment:
This check supersedes the one in Compilation:1849 in the `getPeStream` `Func`. We should remove that try/catch.